app/models/amusement_park.rb

```diff
@@ -1,3 +1,10 @@
 class AmusementPark < ApplicationRecord
   has_many :rides
+
+  def unique_mechanics
+    Mechanic.joins(:ride_mechanics)
```

Comment:
Nice job using multiple `.joins`. Do you need the `.joins(:mechanics)` as well? 

You could also add another association like:
```ruby
has_many :mechanics, through: :rides 
``` 
In your `ride.rb` class you have a:
```ruby
has_many :mechanics, through: :ride_mechanics
```
If you have that additional association, you can now start right at `mechanics` and do something like:
```ruby
mechanics.distinct
```